irs-edc-client/src/test/java/org/eclipse/tractusx/irs/edc/client/policy/service/EdcPolicyDefinitionServiceTest.java

```diff
@@ -127,6 +130,42 @@ void givenPolicy_WhenCreateAccessPolicy_ThenCreateIt() throws CreateEdcPolicyDef
         assertThat(result).isNotBlank();
     }
 
+    @Test
+    void givenPolicyName_WhenGetPolicy_ThenExists() throws GetEdcPolicyDefinitionException {
+        // given
+        when(edcConfiguration.getControlplane()).thenReturn(controlplaneConfig);
+        when(controlplaneConfig.getEndpoint()).thenReturn(endpointConfig);
+        String policyName = "policyName";
+        when(endpointConfig.getPolicyDefinition()).thenReturn("/management/v2/policydefinitions" + "/" + policyName);
+
+        ResponseEntity<String> responseEntity = ResponseEntity.ok("Mocked response");
+        when(restTemplate.getForEntity(anyString(), eq(String.class))).thenReturn(responseEntity);
+
+        // when
+        boolean result = service.policyDefinitionExists(policyName);
+
+        // then
+        assertThat(result).isTrue();
+    }
+
+    @Test
+    void givenPolicyName_WhenGetPolicy_ThenNotExists() throws GetEdcPolicyDefinitionException {
+        // given
+        when(edcConfiguration.getControlplane()).thenReturn(controlplaneConfig);
+        when(controlplaneConfig.getEndpoint()).thenReturn(endpointConfig);
+        String policyName = "policyName";
+        when(endpointConfig.getPolicyDefinition()).thenReturn("/management/v2/policydefinitions" + "/" + policyName);
+
+        ResponseEntity<String> responseEntity = ResponseEntity.badRequest().build();
```

Comment:
Why do you use a Bad Request Code for the case where the policy does not exist? Shouldn't it be a Not Found Code?